Use async_forward_entry_setups instead of async_setup_platforms, update pytest

RobertD502 opened the pull request:
- Use async_forward_entry_setups function instead of async_setup_platforms

* Use of async_setup_platforms will result in the integration failing to start as of Home Assistant 2023.3

RobertD502 commented:
Not sure why the pytest failed. The function [does exist within ConfigEntries](https://github.com/home-assistant/core/blob/28a3b4a32c5234397ebfe6f16b5541bb19ad725a/homeassistant/config_entries.py#L1339).

vasqued2 commented:
Thanks for the heads up.  Lmk if you know of another integration using pytest that has already made the change and I’ll see if I can figure out what’s wrong.

Do you have a link so I can read more about this change in general?

RobertD502 commented:
Sure. It was talked about on the [dev blog](https://developers.home-assistant.io/blog/2022/07/08/config_entry_forwards/)back in July. As of the 2023.2 beta, users are informed in the logs about the change and that integrations using the old method will fail to start beginning with HA 2023.3.

RobertD502 commented:
Just took a look at the workflow and see that an old version of pytest_home_assistant_custom_component is being used:

```

  Downloading pytest_homeassistant_custom_component-0.5.14-py3-none-any.whl (37 kB)
```

The most recent is version `0.12.49`.

RobertD502 commented:
I’ve gone ahead and also added the pytest fixes to this PR.

`Changes to the workflow:`
* Python version set to 3.10 as support for 3.9 has been [removed in Home Assistant 2023.2](https://rc.home-assistant.io/blog/2023/01/25/release-20232#breaking-changes) 
* Upgrade flag set for pip to ensure the latest versions are fetched for the requirements
* Pytest asyncio mode has been set to auto. Prior to this, any asynchronous functions in your tests were ignored by pytest.

`Other changes:`
* Added pytest-asyncio to the requirements

As you can see below, the tests pass now:

![DB15ED00-C774-49F2-9BDB-2DF64D1B0A99](https://user-images.githubusercontent.com/52541649/214785694-974e4b90-010f-4446-aecc-77cff9efc72c.jpeg)

vasqued2 commented:
I really appreciate your help with this as well as the extra cleanup.  It's safe to say this would have abruptly stopped working without your help.  I'll pull it into the next release.  Thanks.

RobertD502 commented:
No problem at all. I figured I'd help some custom integrations that were logged as being affected since I was already doing some work on my own integrations.

Be sure to keep an eye on the developer blog. They are pretty good at informing developers when something is going to break in the future.
